plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -72,65 +80,109 @@ protected void doClose() {
     }
 
     public Pipeline get(String id) {
-        return pipelines.get(id);
+        PipelineReference ref = pipelines.get(id);
+        if (ref != null) {
+            return ref.getPipeline();
+        } else {
+            return null;
+        }
     }
 
     void updatePipelines() {
-        Map<String, Pipeline> pipelines = new HashMap<>();
-        SearchResponse searchResponse = client.prepareSearch(INDEX)
-                .setScroll(TimeValue.timeValueMinutes(1))
-                .addSort("_doc", SortOrder.ASC)
-                .setIndicesOptions(IndicesOptions.lenientExpandOpen())
-                .get();
-        logger.info("Loading [{}] pipelines", searchResponse.getHits().totalHits());
-        do {
-            for (SearchHit hit : searchResponse.getHits()) {
-                logger.info("Loading pipeline [{}] with source [{}]", hit.getId(), hit.sourceAsString());
-                Pipeline.Builder builder = new Pipeline.Builder(hit.sourceAsMap());
-                pipelines.put(hit.getId(), builder.build());
+        int changed = 0;
```

Comment:
that counter was meant to be used in the log statement below so that we can log how many pipelines were updated. But I forgot to use it... I'll use it now.